### PR TITLE
android: fix va_* and cleanup pal + first output

### PR DESCRIPTION
**Pull request by obastemur**
andro-chakra

### CODE REVIEW

**Comment by obastemur** [comment]
@dotnet-bot test Windows x86_release please

**Comment by digitalinfinity** [comment]
Very cool! @EdMaurer FYI

File: pal/inc/pal.h [hunk 1]
```diff
@@ -187,36 +209,12 @@ extern "C" {
 
 /******************* Compiler-specific glue *******************************/
 
-#ifndef _MSC_VER
 #define FEATURE_PAL_SXS 1
```
Comment (digitalinfinity) [inline pal/inc/pal.h]: Why is this change needed?

File: pal/inc/pal.h [hunk 2]
```diff
@@ -187,36 +209,12 @@ extern "C" {
...
-#endif // !_MSC_VER
-
-#if defined(_MSC_VER) || defined(__llvm__)
 #define DECLSPEC_ALIGN(x)   __declspec(align(x))
```
Comment (digitalinfinity) [inline pal/inc/pal.h]: I'm confused- __declspec(align) is a VC++ concept, no? Wasn't the ifdef that was there correct?

File: pal/inc/pal.h [hunk 3]
```diff
@@ -187,36 +209,12 @@ extern "C" {
...
-#else
-#define DECLSPEC_ALIGN(x)
-#endif
-
 #define DECLSPEC_NORETURN   PAL_NORETURN
-
-#ifndef _MSC_VER
 #define __assume(x) (void)0
```
Comment (digitalinfinity) [inline pal/inc/pal.h]: I'm trying to understand the changes to this file- is the idea basically that this file should never be built with VC++? Otherwise, for instance, this change breaks prefast, no?

File: pal/inc/pal.h (hunk 1 above)
Comment (obastemur) [inline pal/inc/pal.h]: Cleanup.

File: pal/inc/pal.h (hunk 2 above)
Comment (obastemur) [inline pal/inc/pal.h]: We use either Clang or VC++.

File: pal/inc/pal.h (hunk 3 above)
Comment (obastemur) [inline pal/inc/pal.h]: > the idea basically that this file should never be built with VC++?

Yes

**Comment by obastemur** [comment]
@digitalinfinity Thanks for the review!!
